app/controllers/UpfrontPaymentController.scala

```diff
@@ -77,46 +81,77 @@ class UpfrontPaymentController @Inject() (
               routes.MonthlyPaymentAmountController.monthlyPaymentAmount()
             }
           journeyService.updateCanPayUpfront(request.journeyId, canPayUpfront)
-            .map(_ => Redirect(pageToRedirectTo))
+            .map(_ => Redirect(pageToRedirectTo.url))
         }
       )
   }
 
-  val upfrontPaymentAmount: Action[AnyContent] = as.authenticatedJourneyAction.async { implicit request =>
-    val mockJourney = MockJourney(userAnswers = UserAnswers.empty.copy(hasUpfrontPayment = Some(true)))
-    Future.successful(Ok(views.upfrontPaymentAmountPage(upfrontPaymentAmountForm(mockJourney), mockJourney.qualifyingDebt, AmountInPence(100L))))
+  val upfrontPaymentAmount: Action[AnyContent] = as.eligibleJourneyAction { implicit request =>
```

Comment:
do we want an action for this endpoint which ensures that the user said they were able to make an upfront payment? Or was the idea to have the actions check as far as the eligibility only?

Reply:
I think eligibility only, could open this up to discussion though